tests/common_docker.py

```diff
@@ -40,19 +40,26 @@ def docker_compose_cmd(arg_list, use_common_files=True):
         for file in COMPOSE_FILES + extra_files:
             files_args += " -f %s" % file
 
-    subprocess.check_call("docker-compose -p %s %s %s" % (conftest.docker_compose_instance, files_args, arg_list), shell=True)
+    cmd = "docker-compose -p %s %s %s" % (conftest.docker_compose_instance, files_args, arg_list)
+    subprocess.check_call(cmd, shell=True)
 
 
 def stop_docker_compose():
     # take down all COMPOSE_FILES and the s3 specific files
     docker_compose_cmd(" -f ../docker-compose.storage.s3.yml -f ../extra/travis-testing/s3.yml down -v")
-
     set_setup_type(None)
 
 
 def start_docker_compose():
     inline_logs = pytest.config.getoption("--inline-logs")
+
+    while os.path.isfile(conftest.lock_file):
+        logging.debug("waiting for lock")
+        time.sleep(1)
+
+    os.mknod(conftest.lock_file)
     docker_compose_cmd("up -d")
+    os.remove(conftest.lock_file)
```

Comment:
Ah yes, that's even better!

But if @GregorioDiStefano is right that xdist may unconditionally kill the subprocess that invokes this, then an actual "flock" might be the only safe alternative, because we will rely on the kernel to clean things up if the killing happens before the unlink is reached.